src/hashset.c

```diff
@@ -1297,9 +1313,9 @@ size_t hashsetScan(hashset *s, size_t cursor, hashsetScanFunction fn, void *priv
             size_t mask_small = expToMask(s->bucket_exp[table_small]);
             size_t mask_large = expToMask(s->bucket_exp[table_large]);
 
-            /* Emit elements in the smaller table, if this bucket hasn't already
-             * been rehashed. */
-            if (table_small == 0 && !cursorIsLessThan(cursor, s->rehash_idx)) {
+            /* Emit elements in the smaller table if it's the new table or if
+             * this bucket hasn't been rehashed yet. */
+            if (table_small == 1 || !cursorIsLessThan(cursor, s->rehash_idx)) {
```

Comment:
The comment is repeating the same logic if the `if` condition. Logic is often easier to understand if it's explained in multiple different ways.

Maybe focus the comment on when it's safe to skip the bucket?

```suggestion
            /* Emit elements in the smaller table. Skip it if it's the old
             * table and this bucket has already been rehashed. */
            if (table_small == 1 || !cursorIsLessThan(cursor, s->rehash_idx)) {
```